### PR TITLE
[FIX] payment_razorpay: invalid email for public users while online payment

### PR DESCRIPTION
**Pull request by paan-odoo**
Steps:
==========
- Install the point_of_sale and Razorpay payment provider modules.
- Create an online payment method for the Razorpay payment provider.
- Add the created payment method to the shop.
- Open the shop, place an order, and use the Razorpay payment method.
- Scan the QR code for payment and complete the payment with public users.

Issue:
==========
- An error occurs stating 'Email must be a valid email address,' causing the task to be blocked.

Cause:
==========
- The email for public users is not found.

FIX:
==========
- Add an empty email address option for public users.

task-3992084

### CODE REVIEW

File: addons/payment_razorpay/models/payment_transaction.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def _razorpay_create_customer(self):
         """
         payload = {
             'name': self.partner_name,
-            'email': self.partner_email,
+            'email': self.partner_email or "",
```
Comment (AntoineVDV) [inline addons/payment_razorpay/models/payment_transaction.py:57]: Same style as the rest.
```suggestion
            'email': self.partner_email or '',
```

**Review by AntoineVDV — APPROVED** [review]
@robodoo delegate+

**Comment by paan-odoo** [comment]
@AntoineVDV 
Changes are done.

Thanks!

**Comment by AntoineVDV** [comment]
@paan-odoo I delegated you the review rights to save time, so you could have replied with @robodoo r+ to merge after the change ;)

@robodoo r+

**Pull request closed without merging**